### DISCUSSION TITLE
关于windows 10 开启了openssh server后， tcp穿透后无法连接上问题

### DISCUSSION THREAD

**Discussion by shangxiaohe** [natfrp/wiki#200] [使用问题] [answered]
问题详描述：
        我在一台windows 10上开启了 openssh server功能后，也在这台电脑上启动了saakuraFrp的客户端，并启动了一条隧道,是穿透tcp的，端口是22，且按文档的进行了认证，然后我用另一台电脑进行 ssh -p 44667 administrator@xxx.xxx.xxx.xxx 进行登录时，一直提示ssh_exchange_identification: Connection closed by remote host 这个错误，百度google了很多方案，都没有用，如果我是通过内网ssh去连接，又是可以连接上的，我不确定是因为内网穿透后还有什么地方没设置好，还是windows 的openssh server功能的问题，有谁遇到过吗

**Comment by fakeboboliu** [comment, marked as answer]
如果需要其他人帮助检查此问题，请连接时使用 `-vvv` 从而提供 debug 日志

**Reply by shangxiaohe** [reply]
shangxiaohe@shangxiaohedeiMac ~ % ssh -vvv administrator@xxx.xxx.xxx.xxx -p 44667
OpenSSH_8.6p1, LibreSSL 3.3.6
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 21: include /etc/ssh/ssh_config.d/* matched no files
debug1: /etc/ssh/ssh_config line 54: Applying options for *
debug2: resolve_canonicalize: hostname xxx.xxx.xxx.xxx is address
debug3: expanded UserKnownHostsFile '~/.ssh/known_hosts' -> '/Users/shangxiaohe/.ssh/known_hosts'
debug3: expanded UserKnownHostsFile '~/.ssh/known_hosts2' -> '/Users/shangxiaohe/.ssh/known_hosts2'
debug1: Authenticator provider $SSH_SK_PROVIDER did not resolve; disabling
debug3: ssh_connect_direct: entering
debug1: Connecting to xxx.xxx.xxx.xxx [xxx.xxx.xxx.xxx] port 44667.
debug3: set_sock_tos: set socket 3 IP_TOS 0x48
debug1: Connection established.
debug1: identity file /Users/shangxiaohe/.ssh/id_rsa type 0
debug1: identity file /Users/shangxiaohe/.ssh/id_rsa-cert type -1
debug1: identity file /Users/shangxiaohe/.ssh/id_dsa type -1
debug1: identity file /Users/shangxiaohe/.ssh/id_dsa-cert type -1
debug1: identity file /Users/shangxiaohe/.ssh/id_ecdsa type -1
debug1: identity file /Users/shangxiaohe/.ssh/id_ecdsa-cert type -1
debug1: identity file /Users/shangxiaohe/.ssh/id_ecdsa_sk type -1
debug1: identity file /Users/shangxiaohe/.ssh/id_ecdsa_sk-cert type -1
debug1: identity file /Users/shangxiaohe/.ssh/id_ed25519 type -1
debug1: identity file /Users/shangxiaohe/.ssh/id_ed25519-cert type -1
debug1: identity file /Users/shangxiaohe/.ssh/id_ed25519_sk type -1
debug1: identity file /Users/shangxiaohe/.ssh/id_ed25519_sk-cert type -1
debug1: identity file /Users/shangxiaohe/.ssh/id_xmss type -1
debug1: identity file /Users/shangxiaohe/.ssh/id_xmss-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_8.6
kex_exchange_identification: Connection closed by remote host
Connection closed by xxx.xxx.xxx.xxx port 44667

这是输出的内容，我对其中一些信息进行了google,但并没有得到帮助

**Reply by fakeboboliu** [reply]
> 这是输出的内容，我对其中一些信息进行了google,但并没有得到帮助

对于你的日志我暂时没有思路，所以请先考虑排除下：
 1. 浏览器打开一下确认有没有通过访问验证
 2. 查看本地端口配置和客户端日志，确保配置正确

**Reply by shangxiaohe** [reply]
1.浏览器打开后提示输入验证密码，输入后显示验证成功，可以关闭浏览器开始连接了，之后再打开这就该网页无法正常运作了，
2.端口也应该是对的上的，如果对不上，正常会在sakuraFrp 客户端的日志中输出连接映射目标127.0.0.1:22失败，请检查本地服务是否打开的，现在没有提示这个

**Reply by fakeboboliu** [reply]
> 1.浏览器打开后提示输入验证密码，输入后显示验证成功，可以关闭浏览器开始连接了，之后再打开这就该网页无法正常运作了，

那请在确定网页无法正常运作的状态下再尝试一下进行 ssh 连接？

**Reply by shangxiaohe** [reply]
还是和上面一样的输出内容，如果我是 ssh administrator@192.168.1.106   这样是可以连接上的，我不知道这样能连接上是不是说明我那台windows 10 的 openssh erver是正常运行且可被连接的，所以我现在并不能确定是哪里的原因导致的

**Reply by fakeboboliu** [reply]
> 还是和上面一样的输出内容，如果我是 ssh administrator@192.168.1.106 这样是可以连接上的，我不知道这样能连接上是不是说明我那台windows 10 的 openssh erver是正常运行且可被连接的，所以我现在并不能确定是哪里的原因导致的

请关闭任何你不认为有必要的选项，比如 **自动 HTTPS**，然后重启隧道

**Reply by shangxiaohe** [reply]
把自动https关闭就可以了，当初新增隧道的时候端口就是选择的SSH，所以认为是默认配制就好了，就没往这方面想，一直以为是公钥私钥的问题，十分感谢大佬的回答

**Reply by fakeboboliu** [reply]
> 所以认为是默认配制就好了

您好，自动https 默认是禁用，除非用户手动选择，不会在任何情况下被自动打开

**Reply by shangxiaohe** [reply]
那就是我手欠，把它给打开来了
